### ISSUE TITLE
Suggested new rule for OBO Foundry ontologies

### ISSUE THREAD

**Issue by leechuck** [OBOFoundry/OBOFoundry.github.io#482]
The OBO Foundry ontologies are not currently mutually interoperable as there are several thousand unsatisfiable classes when all the ontologies are classified within a single model (see https://github.com/bio-ontology-research-group/oboconsistencytest).

We ( @bill-baumgartner @pbuttigieg and I) would suggest to have a new rule for OBO Foundry (not OBO Library) ontologies: classification within a single model without unsatisfiable classes (at least using ELK).

The main way to enforce this constraint would be to incorporate it into the integration checks for OBO Foundry ontologies, i.e., don't perform classification of the ontology and its imports alone, but import it together with all Foundry ontologies (the full ontologies including all their axioms, not a subset thereof). 

Is there any reason why this would not be sensible?

**Comment by pbuttigieg** [comment]
My main concern is that our ENVO assertions may be breaking things in other resources I can't see in our local build process. I'll have to investigate the explanations that will be generated by @leechuck 's code once more compute is available to be able to comment more .

**Comment by balhoff** [comment]
@leechuck I agree and I think we need access to a release of each ontology which includes all axioms, but does not include any pre-reasoned inferences. Many of the release versions of ontologies include a precomputed class hierarchy, which depends on particular versions of imports from external ontologies.

I suggested making available such an artifact for Uberon: https://github.com/obophenotype/uberon/issues/1363

It would also be nice if there were an OBO convention for how to get the full, "raw", version of a given ontology. E.g. for Uberon I suggested the IRI `http://purl.obolibrary.org/obo/uberon/raw/uberon.owl`.

**Comment by alanruttenberg** [comment]
+1 on reasoning on foundry ontologies taken as a whole, and on suggestion by @balhoff that we add purl conventions for some kinds of artifacts. Even better: Have each version of ontology include annotations referring to other available artifacts, so the information is explicit. Added by the build process, of course.

**Comment by matentzn** [comment]
In practice, we do now import other ontologies and classify our ontologies with them. I think this is a good step forward; I would suggest a slightly reduced proposal here:

1. Every ontology SHOULD provide a [base](https://github.com/INCATools/ontology-development-kit/blob/master/docs/ReleaseArtefacts.md#release-artefact-1-base-required)
2. Every ontology primary release MUST be consistent and coherent
3. Every ontology MUST be coherent when classified together with RO, BFO, and COB
4. Every ontology SHOULD be coherent when classified together with all its base-dependencies
 
What do you think?

**Comment by nataled** [comment]
Thanks, @matentzn! So, condensing all that down, I surmise that an ontology base:
- Contains ONLY terms that are owned by the ontology, which include (a) terms using that ontology's prefix; and (b) any terms with a different prefix that are now maintained by that ontology. It does not contain any imports.
- Retains any references to external ontologies within axioms that are needed for definitions and reasoning (I believe such references, when displayed in Protege, will appear as identifiers only; that is, without labels)
- Has had reasoning applied

I know I'm missing things like metadata and annotations, but just keeping it simple for now. Let me know if I got anything wrong.
